### ISSUE TITLE
Godiva3 calendar does not limit temporal bounds

### ISSUE THREAD

**Issue by pacioos** [Reading-eScience-Centre/edal-java#167]
I have noticed in Godiva3 (via TDS5) that the calendar utility does not limit the selectable temporal bounds to those of the dataset being viewed, like was done in TDS4/Godiva2. For example, our "tide_mhi" Hawaii tide model dataset is only available from 2023-12-01T00:00:00Z through 2024-12-31T23:00:00Z:

- **TDS5:** [http://pae-paha.pacioos.hawaii.edu/thredds-dev/catalog/tide_mod.html?dataset=tide_mhi](http://pae-paha.pacioos.hawaii.edu/thredds-dev/catalog/tide_mod.html?dataset=tide_mhi)
- **Godiva3:** [http://pae-paha.pacioos.hawaii.edu/thredds-dev/Godiva.html?server=http%3A%2F%2Fpae-paha.pacioos.hawaii.edu%2Fthredds-dev%2Fwms%2Ftide_mhi](http://pae-paha.pacioos.hawaii.edu/thredds-dev/Godiva.html?server=http%3A%2F%2Fpae-paha.pacioos.hawaii.edu%2Fthredds-dev%2Fwms%2Ftide_mhi)

The Godiva3 viewer, however, will allow you to select years and months before and after this time range. For a new user, it becomes a fishing expedition trying to figure out how far back a dataset goes (how far in the future a forecast goes). Unavailable dates are only signaled by the grayed-out and italicized calendar dates.

In contrast, the Godiva2 viewer (via TDS4) was smarter about limiting the year, month, and hour pull-down menus to within the dataset's available temporal coverage:

- **TDS4:** [http://pae-paha.pacioos.hawaii.edu/thredds/tide_mod.html?dataset=tide_mhi](http://pae-paha.pacioos.hawaii.edu/thredds/tide_mod.html?dataset=tide_mhi)
- **Godiva2:** [http://pae-paha.pacioos.hawaii.edu/thredds/tide_mod.html?dataset=tide_mhi](http://pae-paha.pacioos.hawaii.edu/thredds/godiva2/godiva2.html?server=http://pae-paha.pacioos.hawaii.edu/thredds/wms/tide_mhi)

This made it much simpler for a user to explore the available time range. Could these smarts be added back into Godiva3, I wonder? Thanks for giving this some thought!

**Comment by guygriffiths** [comment]
I don't think so, no. The toolkit we used to build Godiva3 did not have that ability, otherwise we would have used it, and looking at the docs, it still doesn't (unsurprisingly - it was discontinued a while ago).

Really what's needed is a full rewrite of a UI for ncWMS in modern Javascript, but unfortunately I don't have time for this at the moment.

I've never used the calendar selector anyway, I always click to the left of it to enable date selection from a drop-down list.

**Comment by pacioos** [comment]
OK, thanks anyways, Guy. Maybe someday down the line...

Also, did not know that about clicking to the left of the calendar to enable a drop-down list. That's a handy hidden feature to know about, but few are likely to discover this on their own.
